Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -274,10 +291,17 @@ def network_create_engine_wrapper(self, pstate, initialdlstatus=None):
 
                 atp["ti"] = torrentinfo
                 has_resume_data = resume_data and isinstance(resume_data, dict)
+
+                if has_resume_data is not None:
+                    # we have resume data but somehow its not working (Credit mining case)
```

Comment:
Can you explain in more detail what the purpose of this block of code is? I think I don't understand it completely.